metis/bin/metis_client

```diff
@@ -1222,45 +1222,135 @@ EOT
     end
   end
 
-  class Nuke < Command
-    usage "nuke <local_directory>"
+  class Checksum < Command
+    usage "checksum <local_directory>"
+
+    def parse(args)
+      options[:file] = "md5checksum.txt"
+      parse_opts(args) do |opts|
+        opts.on("-f", "--file FILE", "Recursive folder removal") do |file|
+          options[:file] = file
+        end
+      end
+    end
+
+    def md5_cache
+      @md5_cache ||= {}
+    end
+
+    def load_md5_cache(md5file)
+      md5dir = ::File.dirname(md5file)
+
+      File.readlines(md5file).each do |line|
+        checksum, file_name = line.chomp.split(/\s+/, 2)
+        full_path = Pathname.new(::File.join( md5dir, file_name )).cleanpath.to_s
+        next unless ::File.exists?(full_path)
+
+        md5_cache[full_path] = checksum
+      end
+    end
 
     def run(local_directory)
       # collect all files and folders
       folders, files = Dir[::File.join(local_directory, "**", "*")].partition { |f| ::File.directory?(f) }
 
       puts "Found #{files.count} files in #{folders.count} folders, total size #{byte_format(files.map{|f| ::File.size(f)}.sum)}."
 
-      deleted_files = deleted_folders = 0
-      remaining_size = 0
+      md5files = files.select do |file|
+        ::File.basename(file) == options[:file]
+      end
 
-      unless files.empty?
-        # compute md5sums
-        file_md5s = files.map.with_index do |file,i|
-          print "Computing md5s - #{progress_bar(i.to_f / files.length)} - #{i} of #{files.length}\r"
-          [ file, %x{ md5sum #{file} }.split(' ')[0] ]
-        end.to_h
-        puts "Computing md5s - #{progress_bar(1)} - #{files.length} of #{files.length}\n"
+      md5files.each do |md5file|
+        load_md5_cache(md5file)
+      end
+
+      # compute md5sums
+      files.each.with_index do |file,i|
+        file = Pathname.new(file).cleanpath.to_s
+
+        print "Computing md5s - #{progress_bar(i.to_f / files.length)} - #{i} of #{files.length}      \r"
+
+        next if ::File.basename(file) == options[:file] || md5_cache[file]
```

Comment:
What if the file `md5` has changed since it was last cached in `md5checksum.txt` -- should that recalculate the value? Probably defeats the purpose of caching...

Reply:
Yeah we have to assume the cache is accurate for this to work...